Foundation/NSRegularExpression.swift

```diff
@@ -39,11 +39,39 @@ open class NSRegularExpression: NSObject, NSCopying, NSCoding {
     }
     
     open func encode(with aCoder: NSCoder) {
-        NSUnimplemented()
+        guard aCoder.allowsKeyedCoding else {
+            preconditionFailure("Unkeyed coding is unsupported.")
+        }
+        
+        aCoder.encode(self.pattern._nsObject, forKey: "NS.pattern")
+        aCoder.encode(self.options.rawValue._bridgeToObjectiveC(), forKey: "NS.options")
     }
     
-    public required init?(coder aDecoder: NSCoder) {
-        NSUnimplemented()
+    public required convenience init?(coder aDecoder: NSCoder) {
+        guard aDecoder.allowsKeyedCoding else {
+            preconditionFailure("Unkeyed coding is unsupported.")
+        }
+        
+        guard let pattern = aDecoder.decodeObject(forKey: "NS.pattern") as? NSString,
+            let options = aDecoder.decodeObject(forKey: "NS.options") as? NSNumber else {
+                return nil
+        }
+        
+        do {
+            try self.init(pattern: pattern._swiftObject, options: Options(rawValue: options.uintValue))
+        } catch {
+            return nil
+        }
+    }
+    
+    open override func isEqual(_ object: Any?) -> Bool {
+        if let other = object as? NSRegularExpression {
```

Comment:
Would this be more swifty if it was 'guard let other else { return false } and then have the return self == other at the end of the function?